### PR TITLE
[PULL REQUEST] Add bypasses + a fix

### LINKED ISSUE #40: Link1s
### Domain

https://link1s.com/RCVFvNFy

### Link

https://drive.google.com/file/d/1c_XsEAoepm_UTyNUYzANuGMd-wISnvZk/view

### Version

0.0.0

### What browsers are you seeing the problem on?

Chrome

### What OS are you seeing the problem on?

_No response_

### (Optional) Anything else?

https://rodjulian.com/vienna-vacation-is-the-most-beautiful-place-to-visit/

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #227: shorte.st
### Domain

https://shorte.st

### Link

http://sh.st/qK5Km (NSFW ADS)

### Version

0.0.0

### What browsers are you seeing the problem on?

Firefox

### What OS are you seeing the problem on?

Windows

### (Optional) Anything else?

Yes, you can click skip ad, but there is just so much NSFW content + I have to stare at it for 8 seconds before I can skip it. Also tons of scam ads and a download started without my permission.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by kiraio-moe**
- [X] I made sure there are no unnecessary changes in the code;
- [X] Tested on Chromium (Includes Opera, Brave, Vivaldi, Edge, etc);
- [X] Tested on Firefox.

**Improvements**:
- Add EditorConfig to preserve Code Style.
- Add bypasses for link1s.com (fix #40) & sh.st (fix #227). Tried the best.

### CODE REVIEW

File: src/helpers/dom.js [hunk 1]
```diff
@@ -145,34 +150,46 @@ export function ensureDomLoaded(func) {
...
+        observer.disconnect()
+      })
+    })
+
+    observer.observe(document.documentElement, {
+      attributes: true,
+      childList: true,
+      subtree: true
+    })
+  })
```
Comment (NotAProton) [inline src/helpers/dom.js]: Changes to the awaitElement function can be a separate PR, this will probably break a lot of other bypasses

Comment (kiraio-moe) [reply src/helpers/dom.js]: Sure, I'll separate it.

**Comment by kiraio-moe** [comment]
Don't merge this yet, I want to add more bypasses form here.

**Comment by GITHUBTRASHACCOUNT** [comment]
add mboost.me bypass also and fix linkvertise use bypass.city
all liscense to code MIT




MBOOST BYPASS: 
// @match        https://mboost.me/*/*
// @grant        none
// @license MIT
// ==/UserScript==
 
(function() {
    // Find all script elements on the page
    const scripts = document.getElementsByTagName('script');
 
    // Loop through all script elements
    for (let i = 0; i < scripts.length; i++) {
        const script = scripts[i];
        const scriptContent = script.innerHTML;
 
        // Check if the script contains the targeturl pattern
        if (scriptContent.includes('"targeturl":')) {
            // Extract the link from the script content
            const targeturlRegex = /"targeturl":\s*"(.*?)"/;
            const match = scriptContent.match(targeturlRegex);
 
            if (match && match.length > 1) {
                const redirectLink = match[1];
 
                // Redirect to the extracted link
                window.location.replace(redirectLink);
                break;
            }
        }
    }

LOOTLINK BYPASS: 

  [Working] loot-link.com bypasser (includes all lootlabs.gg link variations)

bypasses loot-link.com quickly and efficiently, saving you time and energy.

// @match        https://loot-link.com/s?*
// @match        https://loot-links.com/s?*
// @match        https://lootlinks.co/s?*
// @match        https://lootdest.info/s?*
// @match        https://lootdest.org/s?*
// @match        https://links-loot.com/s?*
// @match        https://linksloot.net/s?*
// @run-at       document-end
// @downloadURL https://update.greasyfork.org/scripts/483207/%5BWorking%5D%20loot-linkcom%20bypasser%20%28includes%20all%20lootlabsgg%20link%20variations%29.user.js
// @updateURL https://update.greasyfork.org/scripts/483207/%5BWorking%5D%20loot-linkcom%20bypasser%20%28includes%20all%20lootlabsgg%20link%20variations%29.meta.js
// ==/UserScript==

(function () {
    if (typeof p == 'object' && p?.PUBLISHER_LINK) {
        window.location.assign(p['PUBLISHER_LINK']);
    }

**Comment by grant** [comment]
Stop! Don't @ mention me.

**Comment by kiraio-moe** [comment]
> add mboost.me bypass also and fix linkvertise use bypass.city all liscense to code MIT
> 
> MBOOST BYPASS: // @match https://mboost.me/*/* // @grant none // @license MIT // ==/UserScript==
> 
> (function() { // Find all script elements on the page const scripts = document.getElementsByTagName('script');
> 
> ```
> // Loop through all script elements
> for (let i = 0; i < scripts.length; i++) {
>     const script = scripts[i];
>     const scriptContent = script.innerHTML;
> 
>     // Check if the script contains the targeturl pattern
>     if (scriptContent.includes('"targeturl":')) {
>         // Extract the link from the script content
>         const targeturlRegex = /"targeturl":\s*"(.*?)"/;
>         const match = scriptContent.match(targeturlRegex);
> 
>         if (match && match.length > 1) {
>             const redirectLink = match[1];
> 
>             // Redirect to the extracted link
>             window.location.replace(redirectLink);
>             break;
>         }
>     }
> }
> ```
> 
> LOOTLINK BYPASS:
> 
> [Working] loot-link.com bypasser (includes all lootlabs.gg link variations)
> 
> bypasses loot-link.com quickly and efficiently, saving you time and energy.
> 
> // @match https://loot-link.com/s?* // @match https://loot-links.com/s?* // @match https://lootlinks.co/s?* // @match https://lootdest.info/s?* // @match https://lootdest.org/s?* // @match https://links-loot.com/s?* // @match https://linksloot.net/s?* // @run-at document-end // @downloadurl https://update.greasyfork.org/scripts/483207/%5BWorking%5D%20loot-linkcom%20bypasser%20%28includes%20all%20lootlabsgg%20link%20variations%29.user.js // @updateURL https://update.greasyfork.org/scripts/483207/%5BWorking%5D%20loot-linkcom%20bypasser%20%28includes%20all%20lootlabsgg%20link%20variations%29.meta.js // ==/UserScript==
> 
> (function () { if (typeof p == 'object' && p?.PUBLISHER_LINK) { window.location.assign(p['PUBLISHER_LINK']); }

Thanks for the reference.

**Comment by undeadjess** [comment]
@kiraio-moe Are you still working on this?

**Comment by kiraio-moe** [comment]
> @kiraio-moe Are you still working on this?

I don't have time to work on this ATM. Sorry. I'm really busy in my RL job :(

**Comment by danilonovaisv** [comment]
Danilo pull

**Comment by lostdusty** [comment]
I'll close this one then, if you got time & still interested, just let us know.
